oss-internship-2020/guetzli/tests/guetzli_sapi_test.cc

```diff
@@ -0,0 +1,141 @@
+#include "gtest/gtest.h"
+#include "guetzli_sandbox.h"
+#include "sandboxed_api/sandbox2/util/fileops.h"
+#include "sandboxed_api/vars.h"
+
+#include <fstream>
+#include <memory>
+#include <syscall.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <sstream>
+#include <algorithm>
+
+namespace guetzli {
+namespace sandbox {
+namespace tests {
+
+namespace {
+
+constexpr const char* IN_PNG_FILENAME = "bees.png";
+constexpr const char* IN_JPG_FILENAME = "nature.jpg";
+constexpr const char* PNG_REFERENCE_FILENAME = "bees_reference.jpg";
+constexpr const char* JPG_REFERENCE_FILENAME = "nature_reference.jpg";
+
+constexpr int PNG_EXPECTED_SIZE = 38'625;
+constexpr int JPG_EXPECTED_SIZE = 10'816;
+
+constexpr int DEFAULT_QUALITY_TARGET = 95;
+constexpr int DEFAULT_MEMLIMIT_MB = 6000;
+
+constexpr const char* RELATIVE_PATH_TO_TESTDATA =
+  "/guetzli_sandboxed/tests/testdata/";
+
+std::string GetPathToInputFile(const char* filename) {
+  return std::string(getenv("TEST_SRCDIR")) 
+    + std::string(RELATIVE_PATH_TO_TESTDATA)
+    + std::string(filename);
+}
+
+std::string ReadFromFile(const std::string& filename) {
+  std::ifstream stream(filename, std::ios::binary);
+
+  if (!stream.is_open()) {
+    return "";
+  }
+
+  std::stringstream result;
+  result << stream.rdbuf();
+  return result.str();
+}
+
+template<typename Container>
+bool CompareBytesInLenValAndContainer(const sapi::v::LenVal& lenval, 
```

Comment:
`CompareBytes` or `CompareData` name should be enough
No need to specify all types from the function signature in its name :)